docs/concepts/abstractions/controllers/garbage-collection.md

```diff
@@ -50,57 +53,76 @@ metadata:
   ownerReferences:
   - apiVersion: extensions/v1beta1
     controller: true
+    blockOwnerDeletion: true
     kind: ReplicaSet
     name: my-repset
     uid: d9607e19-f88f-11e6-a518-42010a800195
   ...
 ```
 
-## Controlling whether the garbage collector deletes dependents
+## Controlling whether and how the garbage collector deletes dependents
+
+When you delete object, you can specify whether the object's dependents are
+deleted automatically. If you delete an object without deleting its dependents
+automatically, the dependents are said to be *orphaned*. Deleting dependents
+automatically is called *cascading deletion*. Further, there are two modes of
+*cascading deletion*: if the object is deleted immediately and the garbage
+collector then deletes the dependents in the background, it is called
+*background cascading deletion*. In contrast, in *foreground cascading
+deletion*, the object first enters a "deletion in progress" state, where the
+object is still visible via the REST API, its `deletionTimestamp` is set, and
+its metadata.finalizers contains "foregroundDeletion". Then the garbage
+collector deletes the dependents. Once the garbage collector has deleted all
+dependents whose ownerReferences.blockOwnerDeletion=true, it will finally delete
+the object.
+
+To control whether delete dependent objects or delete them in
+foreground/background, set the deleteOptions.propagationPolicy to "Orphan",
+"Foregound", or "Background" respectively.
+
+The default garbage collection policy for controller resources is `orphan`. So unless you specify
+otherwise, dependent objects are orphaned.
 
-When you delete object, you can specify whether the object's dependents
-are deleted automatically. Deleting dependents automatically is called
-*cascading deletion*. If you delete an object without deleting its
-dependents automatically, the dependents are said to be *orphaned*.
+Here's an example that deletes dependents in background:
 
-To delete dependent objects automatically, set the `orphanDependents` query
-parameter to false in your request to delete the owner object.
+```shell
+kubectl proxy --port=8080
+curl -X DELETE localhost:8080/apis/extensions/v1beta1/namespaces/default/replicasets/my-repset -d '{"kind":"DeleteOptions","apiVersion":"v1","propagationPolicy":"Background"}' -H "Content-Type: application/json"
```

Comment:
@steveperry-53 @pwittrock i changed to use the body instead of using the query parameters because the latter doesn't work for propagationPolicy (see https://github.com/kubernetes/kubernetes/issues/34856#issuecomment-287227594)